### PR TITLE
Improved visualization of molecular systems

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

File: brayns/io/MolecularSystemReader.cpp [hunk 1]
```diff
@@ -153,12 +169,38 @@ bool MolecularSystemReader::_loadConfiguration()
...
+    while( std::getline( filePositions, line ))
+    {
+        std::stringstream lineStream( line );
+        size_t id;
+        Vector3f position;
+        lineStream >> id >> position.x() >> position.y() >> position.z();
+
+        _calciumPositions.push_back( position );
+    }
+    filePositions.close();
```
Comment (hernando) [inline brayns/io/MolecularSystemReader.cpp]: At least check that filePositions is good after reading.

**Comment by favreau** [comment]
Retest this please

File: brayns/common/camera/FlyingModeManipulator.cpp [hunk 2]
```diff
@@ -79,17 +81,18 @@ void FlyingModeManipulator::wheel( const Vector2i& /*position*/,
...
 void FlyingModeManipulator::_strafeRight()
 {
-    translate( Vector3f::left() * getMotionSpeed(), true );
+    translate( Vector3f::right() * getMotionSpeed(), true );
 }
 
 void FlyingModeManipulator::_flyForward()
 {
     translate( Vector3f::forward() * getWheelSpeed(), true );
+    BRAYNS_INFO << _camera << std::endl;
```
Comment (tribal-tec) [inline brayns/common/camera/FlyingModeManipulator.cpp]: ?

File: brayns/common/transferFunction/TransferFunction.cpp [hunk 3]
```diff
@@ -27,6 +27,12 @@ namespace brayns
 
 TransferFunction::TransferFunction()
 {
+    for( size_t i=0; i<256; ++i )
+    {
+        _diffuseColors.push_back( Vector3f( 0, i/255.f, 0 ));
```
Comment (tribal-tec) [inline brayns/common/transferFunction/TransferFunction.cpp]: you have a state again :)

File: plugins/engines/optix/OptiXScene.cpp [hunk 4]
```diff
@@ -81,100 +81,137 @@ void OptiXScene::reset()
     Scene::reset();
 
     // Geometry
+    /*
```
Comment (tribal-tec) [inline plugins/engines/optix/OptiXScene.cpp]: sure about all this commented lines?

File: plugins/engines/ospray/ispc/render/SimulationRenderer.ispc [hunk 5]
```diff
@@ -37,7 +37,8 @@ inline varying vec4f getSimulationValue(
     varying float& lightEmission )
 {
     lightEmission = 0.f;
-    varying vec4f color = make_vec4f( 0.f );
+    //varying vec4f color = make_vec4f( 0.f );
```
Comment (tribal-tec) [inline plugins/engines/ospray/ispc/render/SimulationRenderer.ispc]: commented code

File: plugins/extensions/plugins/ZeroEQPlugin.cpp [hunk 6]
```diff
@@ -328,6 +380,54 @@ void ZeroEQPlugin::_sceneUpdated( )
             material->setEmission( m.getLightEmission() );
         }
     }
+
+    // Spheres
+    const auto& remoteSpheres = _remoteScene.getSpheresVector();
```
Comment (tribal-tec) [inline plugins/extensions/plugins/ZeroEQPlugin.cpp]: when do we start using zerobuf/lexis directly? I'm still surprised you don't get bored yourself always writing this code...

**Comment by favreau** [comment]
Retest this please

File: plugins/engines/ospray/ispc/render/SimulationRenderer.ispc (hunk 5 above)
Comment (favreau) [inline plugins/engines/ospray/ispc/render/SimulationRenderer.ispc]: Done

File: brayns/common/transferFunction/TransferFunction.cpp (hunk 3 above)
Comment (favreau) [inline brayns/common/transferFunction/TransferFunction.cpp]: Removed

**Pull request merged**